Source/WebCore/rendering/OrderIterator.h

```diff
@@ -40,24 +40,40 @@ class RenderObject;
     
 class OrderIterator {
 public:
+    using iterator_category = std::forward_iterator_tag;
+    using value_type = RenderBox;
+    using difference_type = std::ptrdiff_t;
+    using pointer = RenderBox*;
+    using reference = RenderBox&;
+    pointer operator*();
+
+    OrderIterator& operator++();
+    // Post Increment
+    OrderIterator operator++(int);
+
+    bool operator==(const OrderIterator& other) const;
+    bool operator!=(const OrderIterator& other) const;
+
+    OrderIterator begin();
+    const OrderIterator end() const;
+
     friend class OrderIteratorPopulator;
 
     explicit OrderIterator(RenderBox&);
-
+    OrderIterator(const OrderIterator&) = default;
     RenderBox* currentChild() const { return m_currentChild; }
-    RenderBox* first();
-    RenderBox* next();
     OrderIterator reverse();
     bool shouldSkipChild(const RenderObject&) const;
 
 private:
     void reset();
-
+    RenderBox* first();
+    RenderBox* next();
     RenderBox& m_containerBox;
     RenderBox* m_currentChild;
 
     using OrderValues = StdSet<int>;
-    OrderValues m_orderValues;
+    std::shared_ptr<OrderValues> m_orderValues;
```

Comment:
Why this change ?

Reply:
Without making m_orderValues shared between the begin and end iterator, the != and == comparison does not work correctly. When the member-wise copies occur to create the begin and end objects, their m_orderValuesIterator member (which is a const iterator to an m_orderValues set) points to different containers even though each container's underlying resources are the same. When the begin iterator reaches the end of the container, comparison of begin's m_orderValuesIterator and end's m_orderValuesIterator always returns false. 
This change forces the sets shared between the begin and end iterators to be the same so the != and == comparison works.

Reply:
This is a bit troubling that the iterator is owning the container 🤔

Reply:
But this is from before this PR anyway

Reply:
Agreed, and the class design forces the Iterator to contain more data members than it needs to just iterate over the container safely vs if the Iterator and the container were separated into two different classes.  It may be worth refactoring this eventually because I don't see a benefit to keeping it this way.